### PR TITLE
Replaces dataview._fetch by dataview.fetch

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Because _fetch wasn't really needed and it made things a bit more confusing (IMO). And, it was being used externally (by deep-insights).

@xavijam what do you think?

There's a corresponding PR in deep-insights.js ([CartoDB/deep-insights.js#1079](https://github.com/CartoDB/deep-insights.js/pull/156))

### CODE REVIEW

File: src/dataviews/dataview-model-base.js [hunk 1]
```diff
@@ -167,8 +159,14 @@ module.exports = Model.extend({
   },
 
   fetch: function (opts) {
+    opts = opts || {};
     this.trigger('loading', this);
-    return Model.prototype.fetch.call(this, opts);
+    return Model.prototype.fetch.call(this, {
+      success: opts.success,
```
Comment (xavijam) [inline src/dataviews/dataview-model-base.js]: What happens if error is defined within opts object?

File: src/dataviews/histogram-dataview-model.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ module.exports = DataviewModelBase.extend({
     if (_.isNumber(this.get('end'))) {
       params.push('end=' + this.get('end'));
     }
-    if (_.isNumber(this.get('bins'))) {
+    if (_.isNumber(this.get('start'))) {
```
Comment (xavijam) [inline src/dataviews/histogram-dataview-model.js]: Boyscout?

File: src/dataviews/dataview-model-base.js [hunk 3]
```diff
@@ -167,8 +159,14 @@ module.exports = Model.extend({
   },
 
   fetch: function (opts) {
+    opts = opts || {};
     this.trigger('loading', this);
-    return Model.prototype.fetch.call(this, opts);
+    return Model.prototype.fetch.call(this, {
```
Comment (xavijam) [inline src/dataviews/dataview-model-base.js]: There are more attributes accepted in the fetch options (`complete` for example). Aren't we missing that?

**Comment by xavijam** [comment]
Fine by me!

File: src/dataviews/histogram-dataview-model.js (hunk 2 above)
Comment (alonsogarciapablo) [inline src/dataviews/histogram-dataview-model.js]: I will revert this. Great catch!

File: src/dataviews/dataview-model-base.js [hunk 4]
```diff
@@ -167,8 +159,13 @@ module.exports = Model.extend({
   },
 
   fetch: function (opts) {
+    opts = opts || {};
     this.trigger('loading', this);
-    return Model.prototype.fetch.call(this, opts);
+    return Model.prototype.fetch.call(this, _.extend(opts, {
```
Comment (alonsogarciapablo) [inline src/dataviews/dataview-model-base.js]: This should do it @xavijam!

Comment (xavijam) [inline src/dataviews/dataview-model-base.js]: :+1:

**Comment by alonsogarciapablo** [comment]
restest this please

**Pull request merged**